docs/optimizing.md

```diff
@@ -0,0 +1,131 @@
+
+```ini
+ansible_user=admin
+ansible_password="SomeSecretPassword"
+```
+
+The `ansible_host` variable can contain one or more APIC hosts separated by a comma. If multiple hosts are defined the plugin will try executing tasks on the hosts in the order listed until one completes or they all fail.
+
+```ini
+single_apic  ansible_host=apic.host
+cluster_apic ansible_host=apic1.host,apic2.host,apic3.host
+```
+
+Signature-based authentication can be specified in the inventory.
+
+```ini
+ansible_httpapi_session_key={'admin': "{{ lookup('file', 'admin.key')}}"}
+```
+
+> [!NOTE]
+> `ansible_httpapi_session_key` takes precedence over `ansible_password`.
+
+> [!TIP]
+> Using signature-based authentication with or without ACI HTTPAPI enabled has the same execution time benefit.
+
+### Full Example Inventory using ACI HTTPAPI plugin
+
+```ini
+[aci]
+single_apic  ansible_host=apic.host
+cluster_apic ansible_host=apic1.host,apic2.host,apic3.host
+
+[aci:vars]
+ansible_user=admin
+ansible_password="SomeSecretPassword"
+ansible_connection=ansible.netcommon.httpapi
+ansible_network_os=cisco.aci.aci
+```
+
+## Using the `suppress_` options
+
+Two options are available to users on all ACI modules to assist with optimizing playbook performance by reducing API calls. These options can increase playbook performance at the expense of suppressing some features of the modules.
+
+### `suppress_previous`
+
+If enabled, a GET call to check previous object state will not be sent before a POST update to APIC.
+
+> [!WARNING]
+> This causes the previous return value to be empty. The previous state of the object will not be checked and POST update calls to APIC will contain all properties specified in the task.
+
+#### `suppress_previous` Aliases
+
+- `no_previous`
+- `ignore_previous`
+
+#### `suppress_previous` Example
+
+```yml
+- hosts: aci
+  gather_facts: no
+
+  tasks:
+  - name: Add a new EPG
+    cisco.aci.aci_epg:
+      tenant: production
+      ap: intranet
+      epg: web_epg
+      description: Web Intranet EPG
+      bd: prod_bd
+      suppress_previous: true
+    delegate_to: localhost
+```
+
+### `suppress_verification`
+
+If enabled, a verifying GET call to check current object state will not be sent after a POST call to APIC.
+
+> [!WARNING]
+> This causes the current return value to be set to the proposed value. The current object state including default values will be unverifiable until another task executes for the same object.
+
+#### `suppress_verification` Aliases
+
+- `no_verification`
+- `no_verify`
+- `suppress_verify`
+- `ignore_verify`
+- `ignore_verification`
+
+#### `suppress_verification` Example
+
+```yml
+- hosts: aci
+  gather_facts: no
+
+  tasks:
+  - name: Add a new EPG
+    cisco.aci.aci_epg:
+      tenant: production
+      ap: intranet
+      epg: web_epg
+      description: Web Intranet EPG
+      bd: prod_bd
+      suppress_verification: true
+    delegate_to: localhost
```

Comment:
If delegate_to: localhost is used in the task, Ansible uses the local connection and not HTTPAPI plugin. I suggest we remove it from here.

Reply:
Sure, all done.